android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java

```diff
@@ -164,4 +185,33 @@ private String refineUrl(String url) {
                 return AssetUtils.httpifyIpfsUrl(url);
         }
     }
+
+    @Override
+    public void onActiveOriginChanged(OriginInfo originInfo) {}
```

Comment:
I would suggest to follow that technique for observers https://github.com/brave/brave-core/blob/master/android/java/org/chromium/chrome/browser/crypto_wallet/observers/KeyringServiceObserverImpl.java.
Make an Impl class and pass a delegate to it, leave all empty methods you don't need with a default implementation. Ideally we would like to have a destroy() method in Impl that nulls the delegate so we have `this` object get cleaned by a garbage collector. What we achieve with it:
1. We make classes cleaner as we don't do stub functions like here
2. We can reuse it with much less code in the future
3. There is an issue with closing observers mojo channels in Java, I'm still trying to understand on how to clean it, as basically it doesn't get cleaned even if we try to destroy an object. We pass `this` on `addObserver` and mojo holds that class forever. In that particular case `PortfolioModel` never gets destroyed even if the class that creates it is destroyed. It's not a big deal in that particular case, but we had much bigger problems before I moved them into Impl as we passed activities as `this` to `addObserver` and they never got destroyed when we got out from wallet and we had memory leaks and warnings that we create duplicated activities each time we enter wallet.

Reply:
Implemented a wrapper but with `weakreference`
https://github.com/brave/brave-core/pull/16768/commits/eed62c27ee607f5ef52be28ff9104234bedf30b1:
1. I used `WeakReference` so that we don't have to manually call destroy and android OS can better maintain this to avoid leaks.
2. I like the default interface but since these service observers often change so It's better to have these methods visible though we can anytime add that default interface methods later.